components/browser/state/src/main/java/mozilla/components/browser/state/state/MediaState.kt

```diff
@@ -0,0 +1,27 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.state
+
+/**
+ * Details about the last playing media in this tab.
+ *
+ * @property lastMediaUrl - [ContentState.url] when media started playing.
+ * This is not the URL of the media but of the page when media started.
+ * Defaults to "" (an empty String) if media hasn't started playing.
+ * @property lastMediaAccess The last time media started playing in the current web document.
+ * Defaults to [0] if media hasn't started playing.
+ */
+data class MediaState(
+    val lastMediaUrl: String = "",
+    val lastMediaAccess: Long = 0
+)
+
+/**
+ * Check whether this tab has in progress media - any media which started playing in this HTML document,
+ * irrespective of it's current state (eg: playing, paused, stopped).
+ */
+fun TabSessionState.hasMediaPlaying(): Boolean {
```

Comment:
I don't see this method being used anywhere. Can we remove it? If not I would prefer inlining / add to the call site directly, because the name is a bit confusing.

Reply:
Not used on AC but would be useful for clients interested in this.
Thought it would be better to have this logic in AC (though not sure this is this file is the right place where to declare it) rather than have clients know about implementation details and let them decide what the `LastMediaAccessState` represents.

Reply:
Talked offline about this and maybe a better naming would help.
Christian suggested `hasPlayedMedia` for our scenario:
- in that tab, that website
- sometime ago
- media started playing
- but it's not clear what's the status now.

I see we also have a `components.feature.tabs.ext` package which seems like a more appropriate location for this extension so I'll move it there.